notifier/notifier.go

```diff
@@ -530,7 +530,7 @@ func (n *Manager) sendAll(alerts ...*Alert) bool {
 				n.metrics.sent.WithLabelValues(url).Add(float64(len(alerts)))
 
 				wg.Done()
-			}(ams.client, am.url().String())
+			}(ctx, cancel, ams.client, am.url().String())
```

Comment:
I do not think there is a race with ctx and cancel. Why do you think so? They are scoped to the inner loop.

Reply:
^ same. Even in https://github.com/prometheus/prometheus/pull/6907